static/js/emoji_picker.js

```diff
@@ -748,19 +748,6 @@ export function register_click_handlers() {
         toggle_emoji_popover(this, message_id);
     });
 
-    $("body").on("click", ".actions_popover .reaction_button", (e) => {
-        const message_id = $(e.currentTarget).data("message-id");
-        e.preventDefault();
-        e.stopPropagation();
-        // HACK: Because we need the popover to be based off an
-        // element that definitely exists in the page even if the
-        // message wasn't sent by us and thus the .reaction_hover
-        // element is not present, we use the message's
-        // .fa-chevron-down element as the base for the popover.
-        const elem = $(".selected_message .actions_hover")[0];
-        toggle_emoji_popover(elem, message_id);
-    });
```

Comment:
Why do we get to remove this rather than needing to port it?

Reply:
I noticed while testing that this logic is buggy; if you click on the message actions menu for a different message than the one that is currently selected, and then click on "Add emoji reaction", the popover will appear attached to the selected message, not the one you clicked on. It's clear from the logic this was likely a preexisting bug, though, so perhaps you can just fix that as a follow up.

Reply:
I guess since this isn't ready to merge yet (see comments below), be encouraged to add a commit fixing this bug to this PR.

Reply:
Ok, fixed.

Reply:
Looks great!